tests/TestCase/ORM/AssociationTest.php

```diff
@@ -475,16 +475,35 @@ public function testFinderInConstructor(): void
         $this->assertSame('published', $assoc->getFinder());
     }
 
-    /**
-     * Tests that the defined custom finder is used when calling find
-     * in the association
-     */
-    public function testCustomFinderIsUsed(): void
+    public function testCustomFinderWithTypedArgs(): void
     {
-        $this->association->setFinder('published');
+        $this->association->setFinder('publishedWithArgOnly');
+        $this->assertEquals(
+            ['custom', 'this' => 'custom'],
+            $this->association->find(null, 'custom')->getOptions()
+        );
+        $this->assertEquals(
+            ['what' => 'custom', 'this' => 'custom'],
+            $this->association->find(null, what: 'custom')->getOptions()
+        );
+        $this->assertEquals(
+            ['what' => 'custom', 'this' => 'custom'],
+            $this->association->find(what: 'custom')->getOptions()
+        );
+    }
+
+    public function testCustomFinderWithOptions(): void
```

Comment:
This tests that BC is maintained for old style finder like `TableTest::findWithOptions(SelectQuery $query, array $options)`.

Reply:
I added a test for covering `find('list')` usage as well.